cfme/physical/physical_switch.py

```diff
@@ -0,0 +1,231 @@
+# -*- coding: utf-8 -*-
+"""A model of an Infrastructure PhysicalSwitch in CFME."""
+import attr
+
+from navmazing import NavigateToAttribute
+from cached_property import cached_property
+
+from cfme.common import PolicyProfileAssignable, Taggable
+from cfme.common.physical_switch_views import (
+    PhysicalSwitchDetailsView,
+    PhysicalSwitchesView
+)
+from cfme.exceptions import ItemNotFound
+from cfme.modeling.base import BaseEntity, BaseCollection
+from cfme.utils.appliance.implementations.ui import CFMENavigateStep, navigate_to, navigator
+from cfme.utils.log import logger
+from cfme.utils.pretty import Pretty
+from cfme.utils.providers import get_crud_by_name
+from cfme.utils.update import Updateable
+from cfme.utils.varmeth import variable
+from cfme.utils.wait import wait_for
+
+
+@attr.s
+class PhysicalSwitch(BaseEntity, Updateable, Pretty, PolicyProfileAssignable, Taggable):
+    """Model of an Physical Switch in cfme.
+
+    Args:
+        name: Name of the physical Switch.
+
+    Usage:
+
+        myswitch = PhysicalSwitch(name='sample_switch')
+        myswitch.create()
+
+    """
+    pretty_attrs = ['name']
+
+    name = attr.ib()
+    provider = attr.ib(default=None)
+
+    INVENTORY_TO_MATCH = ['power_state', 'part_number', 'serial_number', 'description',
+                          'manufacturer']
+
+    def load_details(self, refresh=False):
+        """To be compatible with the Taggable and PolicyProfileAssignable mixins.
+
+        Args:
+            refresh (bool): Whether to perform the page refresh, defaults to False
+        """
+        view = navigate_to(self, "Details")
+        if refresh:
+            view.browser.refresh()
+            view.flush_widget_cache()
+
+    def _execute_button(self, button, option, handle_alert=False):
+        view = navigate_to(self, "Details")
+        view.toolbar.custom_button(button).item_select(option, handle_alert=handle_alert)
+        return view
+
+    def _execute_action_button(self, button, option, handle_alert=True, **kwargs):
+        target = kwargs.get("target", None)
+        provider = kwargs.get("provider", None)
+        desired_state = kwargs.get("desired_state", None)
+
+        view = self._execute_button(button, option, handle_alert=handle_alert)
+
+        if desired_state:
+            self._wait_for_state_change(desired_state, target, provider, view)
+        elif handle_alert:
+            wait_for(
+                lambda: view.flash.is_displayed,
+                message="Wait for the handle alert to appear...",
+                num_sec=5,
+                delay=2
+            )
+
+    @variable(alias='ui')
```

Comment:
**REQUIRED:** Unless there are other methods implemented here, we can drop the decorator on all these. This is actually an older way of implementing variable implementation context stuff. We have sentaku for this now and we hope to implement that in other places as time goes on.

Reply:
Done.